src/hyperlink/test/test_url.py

```diff
@@ -1112,7 +1112,6 @@ def test_autorooted(self):
 
         attempt_unrooted_absolute = URL(host="foo", path=['bar'], rooted=False)
         normal_absolute = URL(host="foo", path=["bar"])
-        attempted_rooted_replacement = normal_absolute.replace(rooted=True)
```

Comment:
@glyph: here

Reply:
weird. that's a real bug. we should be asserting something about that object

Reply:
I think that asserting something about that object is unrelated to this PR, though, and is an example of why landing this PR would be good :-)

Reply:
Not sure how you want to address that.  Can you open a PR with a fix or an issue to track it?

Reply:
For reference, this came via #91

Reply:
Yeah, I guess what I meant was *I* should have asserted something about that object :).

You can feel free to just clean it up here for the time being, and assign any resulting issue to me

Reply:
Issue #99 for replacing this.